botorch/generation/gen.py

```diff
@@ -47,7 +47,7 @@ def gen_candidates_scipy(
     upper_bounds: Optional[Union[float, Tensor]] = None,
     inequality_constraints: Optional[List[Tuple[Tensor, Tensor, float]]] = None,
     equality_constraints: Optional[List[Tuple[Tensor, Tensor, float]]] = None,
-    nonlinear_inequality_constraints: Optional[List[Callable]] = None,
+    nonlinear_inequality_constraints: Optional[List[Tuple[Callable, bool]]] = None,
```

Comment:
Thinking about this a bit more, I am somewhat concerned that this backward-incompatible change may cause a bunch of breakages for users. One thing we could do is both accept the old and new format of the constraints (defaulting to the previous intrapoint behavior) and emit a `DeprecationWarning` if the old one is used (we'd keep the type as only the new one to indicate that that is the correct one to use). You could write a little helper function for this that checks the format, converts to the new one and raises a warning if it's the old one - you can then use that helper in different places to avoid code duplication.

That said, this is pretty advanced functionality so I don't expect a ton of people to actually use this, and so we might just be ok hard breaking this...

@saitcakmak any thoughts on this?

Reply:
Good point, so we accept the old format and translate it to Tuple[Callable, bool], where the bool is set to `True` indicating an intrapoint constraint. Only question would be up to which level, we are supporting this. I would recommend only up to `optimize_acqf` and `gen_candidates_scipy`. Not deeper. Ok?

Reply:
It is also a good idea, because than I do not break BoFire, where I currently need only the intra ones ;)

Reply:
>  would recommend only up to optimize_acqf and gen_candidates_scipy. Not deeper. Ok?

Makes sense to me!

Reply:
Done.